### ISSUE TITLE
Cannot create a table with an external S3 location when using an IAM role

### ISSUE THREAD

**Issue by alexferl** [prestodb/presto#15621]
I am trying to run the following query:
```
CREATE TABLE my_table ( 
    field varchar
) WITH ( 
    format = 'parquet',
    external_location = 's3a://my-bucket/my/path/'
);
```

and I am getting the following error:

```
presto:default> CREATE TABLE my_table ( field varchar ) WITH ( format = 'parquet', external_location = 's3a://my-bucket/my/path' );
Query 20210119_224422_00003_s8xw7 failed: 1 validation error detected: Value 'presto-s3a://my-bucket/my/path' at 'roleSessionName' failed to satisfy constraint: Member must satisfy regular expression pattern: [\w+=,.@-]* (Service: AWSSecurityTokenService; Status Code: 400; Error Code: ValidationError; Request ID: 3124f73b-7b89-44b8-b546-afbadfe2bb34)
com.amazonaws.services.securitytoken.model.AWSSecurityTokenServiceException: 2 validation errors detected: Value 'presto-s3a://my-bucket/my/path' at 'roleSessionName' failed to satisfy constraint: Member must have length less than or equal to 64; Value 'presto-s3a://my-bucket/my/path' at 'roleSessionName' failed to satisfy constraint: Member must satisfy regular expression pattern: [\w+=,.@-]* (Service: AWSSecurityTokenService; Status Code: 400; Error Code: ValidationError; Request ID: 3ecb3679-4da6-4dc1-89ed-5d707af6da21)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1381)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1127)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:784)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:745)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.doInvoke(AWSSecurityTokenServiceClient.java:1719)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.invoke(AWSSecurityTokenServiceClient.java:1686)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.invoke(AWSSecurityTokenServiceClient.java:1675)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.executeAssumeRole(AWSSecurityTokenServiceClient.java:589)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.assumeRole(AWSSecurityTokenServiceClient.java:561)
	at com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider.newSession(STSAssumeRoleSessionCredentialsProvider.java:321)
	at com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider.access$000(STSAssumeRoleSessionCredentialsProvider.java:37)
	at com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider$1.call(STSAssumeRoleSessionCredentialsProvider.java:76)
	at com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider$1.call(STSAssumeRoleSessionCredentialsProvider.java:73)
	at com.amazonaws.auth.RefreshableTask.refreshValue(RefreshableTask.java:257)
	at com.amazonaws.auth.RefreshableTask.blockingRefresh(RefreshableTask.java:213)
	at com.amazonaws.auth.RefreshableTask.getValue(RefreshableTask.java:154)
	at com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider.getCredentials(STSAssumeRoleSessionCredentialsProvider.java:299)
	at com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider.getCredentials(STSAssumeRoleSessionCredentialsProvider.java:36)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.getCredentialsFromContext(AmazonHttpClient.java:1239)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:815)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:765)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:745)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5052)
	at com.amazonaws.services.s3.AmazonS3Client.getBucketRegionViaHeadRequest(AmazonS3Client.java:5840)
	at com.amazonaws.services.s3.AmazonS3Client.fetchRegionFromCache(AmazonS3Client.java:5813)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5036)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4998)
	at com.amazonaws.services.s3.AmazonS3Client.getObjectMetadata(AmazonS3Client.java:1335)
	at com.amazonaws.services.s3.AmazonS3Client.getObjectMetadata(AmazonS3Client.java:1309)
	at com.facebook.presto.hive.s3.PrestoS3FileSystem.lambda$getS3ObjectMetadata$4(PrestoS3FileSystem.java:643)
	at com.facebook.presto.hive.RetryDriver.run(RetryDriver.java:139)
	at com.facebook.presto.hive.s3.PrestoS3FileSystem.getS3ObjectMetadata(PrestoS3FileSystem.java:640)
	at com.facebook.presto.hive.s3.PrestoS3FileSystem.getS3ObjectMetadata(PrestoS3FileSystem.java:624)
	at com.facebook.presto.hive.s3.PrestoS3FileSystem.getFileStatus(PrestoS3FileSystem.java:346)
	at org.apache.hadoop.fs.HadoopExtendedFileSystem.getFileStatus(HadoopExtendedFileSystem.java:384)
	at org.apache.hadoop.fs.FileSystem.isDirectory(FileSystem.java:1439)
	at com.facebook.presto.hive.HiveMetadata.getExternalPath(HiveMetadata.java:1228)
	at com.facebook.presto.hive.HiveMetadata.createTable(HiveMetadata.java:934)
	at com.facebook.presto.spi.connector.classloader.ClassLoaderSafeConnectorMetadata.createTable(ClassLoaderSafeConnectorMetadata.java:346)
	at com.facebook.presto.metadata.MetadataManager.createTable(MetadataManager.java:654)
	at com.facebook.presto.execution.CreateTableTask.internalExecute(CreateTableTask.java:196)
	at com.facebook.presto.execution.CreateTableTask.execute(CreateTableTask.java:83)
	at com.facebook.presto.execution.CreateTableTask.execute(CreateTableTask.java:65)
	at com.facebook.presto.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:187)
	at com.facebook.presto.$gen.Presto_0_245_1_9966d7d____20210119_221742_1.run(Unknown Source)
	at com.facebook.presto.execution.SqlQueryManager.createQuery(SqlQueryManager.java:254)
	at com.facebook.presto.dispatcher.LocalDispatchQuery.lambda$startExecution$5(LocalDispatchQuery.java:114)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

It seems like my S3 path is being appended to the `roleSessionName` because of [this line](https://github.com/prestodb/presto/blob/a24e7e16847fa81a0570f816e5aec97e633ec9cb/presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java#L804). It seems like it was `presto-session` previously, but was changed [here](https://github.com/prestodb/presto/pull/10864#discussion_r254360367). 

Presto 0.245.1
Hive standalone metastore 3.1.2

I am running a standalone Hive metastore backed by MySQL and a Presto coordinator and one worker in Docker containers.

I have the following S3 settings in my `${HIVE_HOME}/conf/metastore-site.xml` in the Hive container:
```
  <property>
    <name>fs.s3a.assumed.role.arn</name>
    <value>arn:aws:iam::xxxxxxxxxxxx:instance-profile/prestodb</value>
  </property>

  <property>
    <name>fs.s3.impl</name>
    <value>org.apache.hadoop.fs.s3a.S3AFileSystem</value>
  </property>

  <property>
    <name>fs.s3n.impl</name>
    <value>org.apache.hadoop.fs.s3a.S3AFileSystem</value>
  </property>
  ```

and the following in my `${PRESTO_HOME}/etc/catalog/hive.properties` on the Presto containers:

```
hive.s3.iam-role=arn:aws:iam::xxxxxxxxxxxx:instance-profile/prestodb
```

It seems like [`fs.s3a.assumed.role.session.name`](https://hadoop.apache.org/docs/current/hadoop-aws/tools/hadoop-aws/assumed_roles.html) would fix this but it's not available in [Presto's Hive connector](https://prestodb.io/docs/0.245.1/connector/hive.html#amazon-s3-configuration).

Is there another way of getting this working?

**Comment by alexferl** [comment]
Changing the [line](https://github.com/prestodb/presto/blob/a24e7e16847fa81a0570f816e5aec97e633ec9cb/presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java#L804) the remove the `+ uri` fixes the issue part I cloned the repo and tried it locally. Probably not the _right_ fix though.

**Comment by ashishtadose** [comment]
Thanks, @alexferl for reporting this. I just verified Hive connector S3 access with I am role is broken due to this change.

We had added URI in roleSessionName for better auditing in AWS access however considering regex limitation of roleSessionName [here](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html ) appending URI or even its attributes such as hostname or path will not be feasible.

We could get role session name from presto configs and in its absence can default to some static session name.
I will PR a fix for this shortly.

CC @zhenxiao
